### ISSUE TITLE
Won't step beyond breakpoint

### ISSUE THREAD

**Issue by PhilMDev** [felixfbecker/vscode-php-debug#314]
PHP version:  7.2.6
XDebug version:  php_xdebug-2.6.1-7.2-vc15-x86_64.dll
Adapter version: 1.12.6

Your launch.json:
```json
{
  // Use IntelliSense to learn about possible attributes.
  // Hover to view descriptions of existing attributes.
  // For more information, visit: https://go.microsoft.com/fwlink/?linkid=830387
  "version": "0.2.0",
  "configurations": [
    {
      "name": "Listen for XDebug",
      "type": "php",
      "request": "launch",
      "port": 9000
    },
    {
      "name": "Launch currently open script",
      "type": "php",
      "request": "launch",
      "program": "${file}",
      "cwd": "${fileDirname}",
      "port": 9000
    }
  ]
}
```
XDebug php.ini config:
```ini
[xdebug]
zend_extension="c:/wamp64/bin/php/php7.2.6/ext/php_xdebug-2.6.1-7.2-vc15-x86_64.dll"

xdebug.remote_enable = 1
xdebug.profiler_enable = off
xdebug.profiler_enable_trigger = Off
xdebug.profiler_output_name = cachegrind.out.%t.%p
xdebug.profiler_output_dir ="c:/wamp64/tmp"
xdebug.show_local_vars=0
```

XDebug logfile (from setting `xdebug.remote_log` in php.ini):  
Adapter logfile (from setting `"log": true` in launch.json):

Code snippet to reproduce: None

```php

```
This is a very recent problem.  The debugger will stop at the breakpoint but will not continue.  Even if I tell it to just continue.  Step Over, Step Into and Step Out do nothing.  Restart works as does Stop.  This has been running like a champ.  I haven't debugged anything for a day or two but no longer than that.  I did restart VCS this morning due to a Win 10 update (ugh) so that may have started the problem.  Perhaps a recent update?

I also have PHPStorm and tried the exact same situation there and it works as expected.  In fact, I was able to find my issue.  So, that does act as a work around for right now but I'd rather keep using VSC and your great extension.

Sorry I don't have more information.  I know this can be annoying to find.  Perhaps others will see this and be able to add to the discussion.

Thank you!

**Comment by crispypants** [comment]
Same here.

**Comment by wshaver** [comment]
Same here. Windows 10, worked great for a while suddenly doesn't. Turned on logging and I can see the this in the logs when I push "step over". Nothing happens.

`-> nextRequest
{ command: 'next',
  arguments: { threadId: 1 },
  type: 'request',
  seq: 31 }
`

**Comment by felixfbecker** [comment]
There is nothing I can do here without even having an XDebug logfile (see the issue template). Even if there was, it is unlikely I would be able to reproduce the problem, because I am not seeing this issue myself. If you care about this, why not debug it? Here are the instructions how you can step through the debugger itself: https://github.com/felixfbecker/vscode-php-debug#contributing

**Comment by felixfbecker** [comment]
You can also download any previous version from the the releases page: https://github.com/felixfbecker/vscode-php-debug/releases
If you find one that works, you can look at what change was done in the following release that broke it, inspect the linked pull request and contact the author.

**Comment by PhilMDev** [comment]
So, I tried your suggestion and attempted to install two previous
versions.  Not sure they ever installed properly since the extension
display still showed 1.12.6.  And, neither worked.  I understand that you
probably can't duplicate.  However, it would've been nice to get this
information earlier.  BTW, I did try the Devsense version (pay for) and it
did work properly.  And, I'm trying to decide if it's worth the $179 to
have a good debugger.  Probably is.

I wish I had the time to get into yours and figure out what's wrong.  I
appreciate all the effort you put into this.

**Comment by felixfbecker** [comment]
That’s funny because if you look at the source code in the devsense vsix you can see that it’s just this extension’s code obfuscated.

**Comment by PhilMDev** [comment]
The instructions from VS Code were not very clear.  It's basically a two
step process.  Turn off the auto update and then install from a file that
you've downloaded.  Then when you reload the editor it doesn't just
overwrite the file you installed.

I tried both 1.12.5 and 1.12.4.  Both failed.  As I said in the original
post, I believe this was a change that was made to VS Code and not your
extension.  Unfortunately, it is effecting your extension.

Have you tried debugging anything?  I suspect any attempt would fail.  I'm
more than willing to create a simple test that I can send to you if you'd
like.

**Comment by bryandandan** [comment]
Were you able to find a solution to this?

**Comment by PhilMDev** [comment]
No.  I've got a copy of PHPStorm and I just use that.

**Comment by joshbouganim** [comment]
i also had this issue but just now i updated vs code to 1.30.0 and the problem disappeared. not sure if they are related

**Comment by PhilMDev** [comment]
So, when I first started programming in 1977 I learned quickly that any problem I was sure was in the language I was using (Basic at that time) was probably with me.  Over the years, making sure I didn't forget that was a helpful way to solve problems I had.  Unfortunately, this time I seem to have forgotten that.

To get to the point, my problem was a variable in the Watch box that didn't exist.  My code would execute a require_once command which looked like it worked just fine but the next command, a curl_init stopped the debugger.  I couldn't get beyond.  I finally noticed that I was getting a error in the Variables -> Locals -> $errstr: "Undefined variable: subaccts".  Of course, I hadn't named that variable in anything I was running so it took me some time to realize that it was the Watch that was causing the problem.

IMHO, this shouldn't have caused the debugger to stop.  However, I suspect this has nothing to do with Felix's code but with VSC.

Bottom line, I found my problem.  Removed the Watch for $subaccts and the debugger works just fine.  A great way to start the New Year.

Hopefully this might help someone else who has run into the same problem.  Make sure you clear your Watches if you no longer need them and especially if the values aren't defined!

Thanks for hanging in there Felix.  Hope you have a great New Year.

**Comment by wshaver** [comment]
Thanks @PhilMDev I removed my watch statements and now the debugger steps properly. I don't know that we should close this issue, as there's still a bug here obviously.

**Comment by PhilMDev** [comment]
@wshaver - Glad this worked for you also!  I think we should let Felix comment on this, if he wants to.  Perhaps the issue should be moved to VSC instead?
